core/src/main/java/org/elasticsearch/action/bulk/BulkItemResponse.java

```diff
@@ -40,7 +44,14 @@
     /**
      * Represents a failure.
      */
-    public static class Failure {
+    public static class Failure implements Writeable<Failure>, ToXContent {
+        static final String INDEX_FIELD = "index";
+        static final String TYPE_FIELD = "type";
+        static final String ID_FIELD = "id";
+        static final String CAUSE_FIELD = "cause";
+        static final String STATUS_FIELD = "status";
+
+        public static final Failure PROTOTYPE = new Failure(null, null, null, null);
```

Comment:
can we just have a `Failure(StreamInput in)` instead of the prototype I don't know where this pattern came from but I see it in a lot of place